### PR TITLE
Contributing a Blog Post on Introduction to Shell Computing - Emily Humphrey

### PR DESCRIPTION
**Pull request by emhumphrey**
Ready for review! I wrote my post on shell computing. @naupaka @melt6457 @mhaliz

### CODE REVIEW

File: _posts/2020-11-06-shell-computing.markdown [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+title:  "Introduction to Shell Computing"
+date:   2020-11-08 12:54:04 -0700
+categories: shell computing
+---
+**What is a shell?**
+When you think of a shell what may first come to mind is the outer covering of an animal such as a crab or a turtle. However, when you leave the animal world and enter into the computer world there is another kind of shell.
+
+In the case of computer science, a shell is the tool a user can utilize in order to interact with the components of a computer. A shell is essentially a software program that is able to interpret commands from the user so the operating system can understand them and perform the various functions. It got the name shell because it is the outermost layer around the operating system. Shells can run on local systems but there is a way to make remote systems available as well making it possible to access remote servers. For example, our server at University of San Francisco, does not have a graphical interface so in this case we need to use the command line to access it in order to do our projects.   
+
+**Command-line interface vs graphical user interface**
```
Comment (DingShen29) [inline _posts/2020-11-06-shell-computing.markdown:12]: Nice explanation for these two types of interface. I would recommend to add a picture for both interface to help beginners virtualize the difference between CLI and GUI.

File: _posts/2020-11-06-shell-computing.markdown [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+**Examples of command line shells**
+There are various shells that one can use such as C shell (csh), Kornshell (ksh), tcsh, and Bash (sh). Each one has its differences and advantages and disadvantages. Shells are like brands. Everyone has a favorite and will religiously defend that choice. Below is a table comparing various command line shells and some of the features they have in comparison to one another.
+
+![shell comparison]({{ site.url }}/assets/shell_functions.png)
+
+**Highlighting Bash**
+The most commonly used shell tends to be the bourne again shell (bash). Bash tends to be popularly used because it is easy to use for beginners. Bash has great features such as tab completed commands. This makes it so you do not have to type out everything all the time. It has the history saved and if you give it the first few letters you can tab and it will auto complete it. This is a huge time saver and also helps cut down on errors you can make while typing. It is important to remember that every shell has slightly different syntax. Below is a list of commonly used commands in bash with a short description of what those commands do.
```
Comment (DingShen29) [inline _posts/2020-11-06-shell-computing.markdown]: You may miss a word after "This makes it".

**Review by DingShen29 — CHANGES_REQUESTED** [review]
Overall nice work! You introduced shell very clearly. I just added two comments, hopefully they are helpful in your post.

**Comment by naupaka** [comment]
@emhumphrey quick adjustment -- your PR title should reflect the content you are hoping to merge. Something like "Contribute a blog post on shell computing"

**Comment by emhumphrey** [comment]
@DingShen29 Thanks for your feedback! I incorporated both changes into my post.

@naupaka Thank you I revised the pr title to reflect the content in my post.

**Review by mhaliz — APPROVED** [review]
Looks good! I'm going to merge these changes now.

**Pull request merged**